packages/fx-core/src/component/driver/script/scriptDriver.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * @author huajiezhang <huajiezhang@microsoft.com>
+ */
+import { assembleError, err, FxError, ok, Result } from "@microsoft/teamsfx-api";
+import { Service } from "typedi";
+import { DriverContext } from "../interface/commonArgs";
+import { ExecutionResult, StepDriver } from "../interface/stepDriver";
+import { exec, ExecException } from "child_process";
+import * as path from "path";
+import fs from "fs-extra";
+import { DotenvOutput } from "../../utils/envUtil";
+
+const ACTION_NAME = "script";
+
+interface ScriptDriverArgs {
+  run: string;
+  workingDirectory?: string;
+  shell?: string;
+  timeout?: number;
+  redirectTo?: string;
+}
+
+@Service(ACTION_NAME)
+export class ScriptDriver implements StepDriver {
+  async run(args: unknown, context: DriverContext): Promise<Result<Map<string, string>, FxError>> {
+    const typedArgs = args as ScriptDriverArgs;
+    const res = await this.executeCommand(typedArgs, context);
+    if (res.isErr()) return err(res.error);
+    const outputs = res.value[1];
+    const kvArray: [string, string][] = Object.keys(outputs).map((k) => [k, outputs[k]]);
+    return ok(new Map(kvArray));
+  }
+  async execute(args: unknown, ctx: DriverContext): Promise<ExecutionResult> {
+    const res = await this.run(args, ctx);
+    return { result: res, summaries: ["run script"] };
```

Comment:
Can the summary be refined to contain other messages? Like whether i's successful or timeout, what command was run.